### ISSUE TITLE
Player location 

### ISSUE THREAD

**Issue by counterfitninja** [bedrock-viz/bedrock-viz#108]
Is it possible to add a function to just updated the player Location so that it could be updated more like a location map rather than a full render each time?

**Comment by cabbey** [comment]
I was thinking about this... what about just doing a geojson update... no tile generation, just that file. I think that would accomplish what you want.

**Comment by counterfitninja** [comment]
That was pretty much my thinking. Not sure how long that would take could you just get the player Location and add them to a seperate file to make it very quick?

**Comment by edward3h** [comment]
Are you looking to get live updates with this? My understanding is that it's not safe for multiple processes to access the leveldb files at the same time i.e. you shouldn't run bedrock-viz against the db of a live server. So the best you can do is take regular backups and run bedrock-viz against those.

I've thought about doing an experiment to run backups every minute against our server and see if players notice, but like many things haven't got around to it yet :(

**Comment by cabbey** [comment]
yeah, trying to access the leveldb while the server is running it is a recipe for disaster. Backing up and using the most recent backup is *always* the best way to go. Or at the very least, ask the server to do a save hold... access the leveldb in *readonly* (I'm not sure bedrock-viz and do that today) then flag the server to resume writes. Which is basically what the backup process does.

Currently, [the scripts I use to manage my server](https://github.com/TapeWerm/MCscripts) can do a backup very quickly without noticeable lag, but it's a bit longer than a minute on my big worlds, so every minute like @edward3h ponders may not be feasible... but I bet I could get away with every 5 minutes. I have them getting archived across the network to an external storage server, which may be hurting as much as it helps.

I suspect the geojson file could be written out pretty fast. If not the whole thing, then at least the players section could be refreshed in an existing one. It might be best to have a separate binary for that, as it's needs reading in the db are quite different, and it might not need to parse the entire thing. Then of course the webapp would need to dynamically update....

**Comment by counterfitninja** [comment]
What I have at the moment is using Windows server snapshots  (previous version) to backup the scripts and not done in game. It's pretty instant. I do this every 15 mins and build my maps off a backup of those.

I had been wondering about putting the game in hold mode for readonly access and getting player data that way but not got enough skills to understand it.
